packages/engine/src/simulation/package/state/packages/behavior_execution/mod.rs

```diff
@@ -231,12 +230,19 @@ impl Package for BehaviorExecution {
     async fn run(&mut self, state: &mut State, context: &Context) -> Result<()> {
         tracing::trace!("Running BehaviorExecution");
         let mut state_proxy = state.write()?;
+        state_proxy.maybe_reload()?;
         let agent_pool = state_proxy.agent_pool_mut();
 
         self.fix_behavior_chains(agent_pool)?;
         self.reset_behavior_index_col(agent_pool)?;
         agent_pool.flush_pending_columns()?;
 
+        // TODO: Have to reload state agent batches twice, because
```

Comment:
Behavior execution could be changed so that they only need to be reloaded once, e.g. by getting the first language before fixing behavior chains using the `behaviors` column strings (instead of reading behavior ids in Rust) or by returning the first language from `fix_behavior_chains`.

Reply:
Me neither, removed "TODO: "

Reply:
Could you instead add my comment after the TODO, if it's clear enough?
(TODO: ... Behavior execution could be changed so that they only need to be reloaded once, e.g. by getting the first language before fixing behavior chains using the behaviors column strings (instead of reading behavior ids in Rust) or by returning the first language from fix_behavior_chains.)

Reply:
Added the TODO